src/huntsman/pocs/dome/musca.py

```diff
@@ -242,34 +242,25 @@ def _get_status_string(self):
             return 'Shutter in ILLEGAL state?'
         return 'Unexpected response from Huntsman Shutter Controller: %r' % v
 
-    def _get_status_dict(self, max_reads=None):
+    def _get_status_dict(self):
         """ Return dictionary of musca status.
-        Args:
-            max_reads (int, optional): Maximum number of lines to read in order to get the status.
-                This is useful because musca seems to send the status of the same component
-                several times consecutively. If None (default, use 10*the number of expected lines).
         Returns:
             dict: The dome status.
         """
-        if max_reads is None:
-            max_reads = 10 * len(Protocol.VALID_DEVICE)
-
         self._write_musca(Protocol.GET_STATUS)  # Automatically sleeps for self._command_delay
 
         status = {}
-        for i in range(max_reads):
-            try:
-                k, v = self.serial.read().strip().split(':')
-            except ValueError:  # not enough values to unpack
-                continue
-            if k == Protocol.SOLAR_ARRAY or k == Protocol.BATTERY:
-                v = float(v)
-            if k != 'Status':
-                status[k] = v
-            if all([v in status.keys() for v in Protocol.VALID_DEVICE]):
-                break
-            elif i == max_reads - 1:
-                raise error.PanError("Max reads reached while attempting to get dome status.")
+        num_lines = len(Protocol.VALID_DEVICE) + 1  # +1 because first line is "Status"
+        for i in range(num_lines):
+
+            raw_response = self.serial.read()
+            response = [s.strip() for s in raw_response.split(":")]
+
+            if response[0] != "Status":
+                # The first line of the status query should begin with "Status"
+                if i == 0:
+                    raise error.PanError("Serial communication problem reading musca status.")
```

Comment:
We should probably get in the habit of raising more specific exceptions.

```suggestion
                    raise error.BadSerialConnection("Serial communication problem reading musca status.")
```